CMakeLists.txt

```diff
@@ -307,58 +307,57 @@ include(cmake/leveldb.cmake)
-    try_append_cxx_flags("-Wsuggest-override" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wimplicit-fallthrough" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wunreachable-code" TARGET warn_interface SKIP_LINK)
-    try_append_cxx_flags("-Wdocumentation" TARGET warn_interface SKIP_LINK)
-
-    # Some compilers (gcc) ignore unknown -Wno-* options, but warn about all
-    # unknown options if any other warning is produced. Test the -Wfoo case, and
-    # set the -Wno-foo case if it works.
-    try_append_cxx_flags("-Wunused-parameter" TARGET warn_interface SKIP_LINK
-      IF_CHECK_PASSED "-Wno-unused-parameter"
-    )
-    try_append_cxx_flags("-Wself-assign" TARGET warn_interface SKIP_LINK
-      IF_CHECK_PASSED "-Wno-self-assign"
-    )
-  endif()
+if(NOT cxx_flags_overridden AND NOT CMAKE_TOOLCHAIN_FILE)
+  try_append_cxx_flags("-Wall" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wextra" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wgnu" TARGET warn_interface SKIP_LINK)
+  # Some compilers will ignore -Wformat-security without -Wformat, so just combine the two here.
+  try_append_cxx_flags("-Wformat;-Wformat-security" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wvla" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wshadow-field" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wthread-safety" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wloop-analysis" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wredundant-decls" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wunused-member-function" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wdate-time" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wconditional-uninitialized" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wduplicated-branches" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wduplicated-cond" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wlogical-op" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Woverloaded-virtual" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wsuggest-override" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wimplicit-fallthrough" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wunreachable-code" TARGET warn_interface SKIP_LINK)
+  try_append_cxx_flags("-Wdocumentation" TARGET warn_interface SKIP_LINK)
+
+  # Some compilers (gcc) ignore unknown -Wno-* options, but warn about all
+  # unknown options if any other warning is produced. Test the -Wfoo case, and
+  # set the -Wno-foo case if it works.
+  try_append_cxx_flags("-Wunused-parameter" TARGET warn_interface SKIP_LINK
+    IF_CHECK_PASSED "-Wno-unused-parameter"
+  )
+  try_append_cxx_flags("-Wself-assign" TARGET warn_interface SKIP_LINK
+    IF_CHECK_PASSED "-Wno-self-assign"
+  )
 endif()
 unset(cxx_flags_overridden)
+if(MSVC)
```

Comment:
Why are we special casing MSVC compared to other compilers, and still turning on warning flags even if cxxflags is overriden, and/or we might be using a toolchain file? Shouldn't this be in the if block above?

Reply:
In the master branch, warning are suppressed for the MSVC builds unconditionally.

However, I agree with your points. This logic is to be reworked and simplified / unified once https://github.com/bitcoin/bitcoin/pull/25972 is merged.

> ... and/or we might be using a toolchain file?

MSVC builds being vcpkg consumers always use a toolchain file provided by vcpkg.